### ISSUE TITLE
[question] Using `full_package_mode` for glib with gtk dependencies causes brittle builds

### ISSUE THREAD

**Issue by hesham-essam** [conan-io/conan-center-index#11684]
My PR for building gtk on windows was successfully built #11308, but after #11545 it can't be built anymore, since harfbuzz updated its glib version. And since gtk and its dependencies use `full_package_mode` for glib (which is necessary to prevent multiple instances of glib), I get `MISSING_DEPENDENCIES` error for `harfbuzz` (as harfbuzz with the old glib is not in the cache anymore).

This hard dependency makes it hard to update glib, since now if glib's version is bumped in any of gtk's dependencies, gtk will fail to build in the CI until all other dependencies bump glib's version as well.

How do you think this should be handled?

**Comment by planetmarshall** [comment]
I'm not sure what the best solution for this is - I'm inclined to suggest static glib should be an unsupported configuration (I can't debug a GTK app in Linux built with static glib - even with everything built from scratch), but I'm not sure that would solve the problem.

This is really an issue with the cached binaries, as there's nothing really wrong with the recipe - it can be built using `--build missing` by the end user with no problems.

**Comment by morevnaproject** [comment]
Hi! Any news on this topic? It would be really nice to have a solution, as we need Gtk. ^__^

**Comment by planetmarshall** [comment]
I think this is going to continue to be a problem for any recipes where `full_package_mode` is specified. Is it possible for CCI packages to be built with `--build missing`? Or would that just consume too many resources?

**Comment by uilianries** [comment]
CCI will not use `--build missing`.  Will need to continue to update recipes.

**Comment by hesham-essam** [comment]
Now that glib is shared, I think we should be removing`full_package_mode` (or at least use it only when linking against static glib). These patches are really causing problems.

**Comment by uilianries** [comment]
@hesham-essam Yes, removing full_package_mode sounds correct. We will need new PRs with those fixes.

**Comment by hesham-essam** [comment]
This is solved now
